src/thor/bidirectional_astar.cc

```diff
@@ -402,10 +402,17 @@ void BidirectionalAStar::Expand(baldr::GraphReader& graphreader,
   uint32_t shortcuts = 0;
 
   // Update the time information even if time is invariant to account for timezones
-  auto seconds_offset = invariant ? 0.f : pred.cost().secs;
-  auto offset_time = FORWARD
-                         ? time_info.forward(seconds_offset, static_cast<int>(nodeinfo->timezone()))
-                         : time_info.reverse(seconds_offset, static_cast<int>(nodeinfo->timezone()));
+  auto seconds_offset = (invariant || !time_info.valid) ? 0.f : pred.cost().secs;
+  TimeInfo offset_time;
+
+  // either we're expanding the reverse tree and the date time type is arrive_by,
+  // or we're expanding the reverse tree and the time type is not arrive_by,
+  // in either case, we're on the time aware branch and so we need to update the time info
+  if constexpr (FORWARD) {
+    offset_time = time_info.forward(seconds_offset, static_cast<int>(nodeinfo->timezone()));
+  } else {
+    offset_time = time_info.reverse(seconds_offset, static_cast<int>(nodeinfo->timezone()));
+  }
```

Comment:
This actually doesn't change at all, only the offset seconds may be 0 if the time isn't valid in the first place.